src/test/java/helpers/WaitUtils.java

```diff
@@ -51,16 +54,38 @@ public void waitForElementInvisibility(long timeout, By by) {
         wait.until(ExpectedConditions.invisibilityOfElementLocated(by));
     }
 
-    public void waitForElementInvisibilityAfterShortWait(By by){
+    public void waitForElementInvisibilityAfterShortWait(By by) {
         waitForElementInvisibility(WaitTimes.SHORT.getValue(), by);
     }
 
-    public void waitForElementPresence(long timeout, By by){
+    public void waitForElementInvisibilityAfterMiddleWait(By by) {
+        waitForElementInvisibility(WaitTimes.MIDDLE.getValue(), by);
+    }
+
+    public void waitForDataLoading(By by) {
+        try {
+            driver.findElement(by).isDisplayed();
+            waitForElementInvisibilityAfterMiddleWait(by);
+        } catch (NoSuchElementException e) {
+            e.printStackTrace();
```

Comment:
if (driver.IsElementDisplayed(by))
            {
                try
                {
                    driver.WaitForElementsToBeNotDisplayed(by, waitTime);
                }
                catch (Exception e)
                {
                    throw new Exception(
                        $"Loading spinner is displayed longer that {waitTime.GetValue()} seconds: {driver.Url}. {e}");
                }